data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/s3/S3FileReader.java

```diff
@@ -61,18 +70,12 @@ private void validateFileType(final String filePath) {
         }
     }
 
-    private void validateS3ObjectSize(final URI fileUri) {
-        final GetObjectAttributesRequest getObjectAttributesRequest = GetObjectAttributesRequest.builder()
-                .bucket(fileUri.getHost())
-                .key(fileUri.getPath().substring(1))
-                .objectAttributes(ObjectAttributes.OBJECT_SIZE)
-                .build();
-
-        final GetObjectAttributesResponse objectAttributes = s3Client.getObjectAttributes(getObjectAttributesRequest);
-        final Long objectSize = objectAttributes.objectSize();
+    private void validateS3ObjectSize(final ResponseInputStream<GetObjectResponse> responseInputStream, final URI uri) {
+        final Long contentLength = responseInputStream.response().contentLength();
 
-        if (objectSize > MAX_FILE_SIZE) {
-            throw new S3ObjectTooLargeException(String.format("S3 Object size can't be more than 5MB. %s object size is %s", fileUri.getPath().substring(1), objectSize));
+        if (contentLength > MAX_FILE_SIZE) {
+            throw new S3ObjectTooLargeException(String.format("S3 object content length can't be more than 5MB. %s object size is %s bytes",
```

Comment:
Here the hard-coded 5MB should be replaced by MAX_FILE_SIZE.

Also, I think MAX_FILE_SIZE should be changed from`5_000_000L` to `5L << 20`?

Reply:
@kkondaka I have never used the bit shift operators in Java, not sure if it's the convention we want to follow here. And I don't think `5L << 20` is equal `5_000_000L`, correct me if I'm wrong.

Reply:
The note about hard-coding "5MB" here is still not addressed:.

> S3 object content length can't be more than 5MB

If we change the constant value to 5_000_000L to 6_000_000L we might miss that string "5MB" and it would be inaccurate.